packages/accounts-controller/src/AccountsController.ts

```diff
@@ -1160,6 +1135,41 @@ export class AccountsController extends BaseController<
     return accountsState;
   }
 
+  /**
+   * Handles the change in multichain network by updating the selected account.
+   *
+   * @param args - The arguments to handle the multichain network change.
+   * @param args.evmClientId - The ID of the EVM client.
+   * @param args.nonEvmChainId - The CAIP2 of the non-EVM chain.
+   */
+  readonly #handleMultichainNetworkChange = ({
+    evmClientId,
+    nonEvmChainId,
+  }: {
+    evmClientId?: string;
+    nonEvmChainId?: CaipChainId;
+  }) => {
+    let accountId: string;
+
+    if (nonEvmChainId) {
+      // Update selected account to non evm account
+      const lastSelectedNonEvmAccount =
+        this.getSelectedMultichainAccount(nonEvmChainId);
+      // @ts-expect-error - This should never be undefined, otherwise it's a bug that should be handled
```

Comment:
Should we throw a specific error, so that if it is a bug, we have some more information?

Reply:
@montelaidev  Are there any plans to fix for alleviating the issue?

 I've had issues testing for this since it's throwing within a private function that is hooked up to a subscription. Do you have a suggestion on how to properly test for the throw or should I just lower the testing threshold?

Reply:
Will defer to accounts team to resolve on accounts controller since this shouldn't be possible